src/main/java/io/strimzi/CertificateWatch.java

```diff
@@ -0,0 +1,168 @@
+    private static final Logger LOG = LoggerFactory.getLogger(CertificateWatch.class);
+
+    private final boolean enabled;
+
+    private ChangeWatcher watcher;
+
+    /**
+     * Constructs the certificate watch. The Watcher class is created only if the certificate watch is enabled.
+     */
+    public CertificateWatch() {
+        this.enabled = ConfigProvider.getConfig().getOptionalValue("certificate.watch.enabled", Boolean.class).orElse(false);
+
+        if (enabled) {
+            try {
+                String path = ConfigProvider.getConfig().getOptionalValue("certificate.watch.path", String.class).orElse("/etc/webhook-certificates/");
+
+                LOG.info("Creating certificate watch for path {}", path);
+
+                this.watcher = new ChangeWatcher(path, () -> {
+                    LOG.info("Certificate change detected => Drain Cleaner will be restarted");
+                    Quarkus.asyncExit(0);
+                });
+            } catch (IOException e) {
+                LOG.error("Failed to create certificate watch", e);
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    /**
+     * Starts the watcher (if enabled)
+     *
+     * @param ev    Startup event
+     */
+    void onStart(@Observes StartupEvent ev) {
+        if (enabled) {
+            LOG.info("Starting certificate watch");
+            watcher.start();
+        } else {
+            LOG.info("Certificate watch is not enabled");
+        }
+    }
+
+    /**
+     * Stops the watcher (if enabled)
+     *
+     * @param ev    Shutdown event
+     */
+    void onStop(@Observes ShutdownEvent ev) {
+        if (enabled) {
+            LOG.info("Stopping certificate watch");
+            try {
+                watcher.stop();
+            } catch (InterruptedException e) {
+                LOG.warn("Failed to stop the certificate watcher", e);
+            }
+        }
+    }
+
+    /**
+     * Internal class which does the watching of a directory for changes using the Java NIO Watch service. If any change
+     * is detected, a handler is called to take the action.
+     */
+    public static class ChangeWatcher implements Runnable {
+        private final WatchService watchService;
+        private final Thread watcherThread;
+        private final Runnable changeHandler;
+
+        private volatile boolean stop = false;
+
+        /**
+         * Constructs the ChangeWatcher to watch for changes to a directory on the filesystem
+         *
+         * @param path      The directory which should be watched
+         * @param handler   The handler which should be run when any change is detected. The handler allows us to plug-in
+         *                  a custom behavior to react for the file change.
+         *
+         * @throws IOException  IOException is thrown if we fail to create the Watch service
+         */
+        public ChangeWatcher(String path, Runnable handler) throws IOException {
```

Comment:
`Runnable` doesn't require that an implementation supports multiple calls to `run()` but the contract you're providing here allows calling it multiple times. Perhaps it's better to just declare your own `@FunctionalInterface`-annotated interface with an `onChange()` method.